cloudrun_locals.tf

```diff
@@ -40,12 +40,36 @@ locals {
       key => lookup(local.cloudrun_iam[key], "bindings", {})
   }
 
-//  cloudrun_traffic = local.cloudrun_traffic_config == {} ? {} : {
   cloudrun_traffic = {
     for service, specs in local.cloudrun_specs: service => {
       for revision, percent in local.cloudrun_traffic_config: replace(revision, ";", "-") => percent
       if length(regexall("^${service};", revision)) > 0
     }
   }
+
+  cloudrun_secrets_attach = {
+    for service, specs in local.cloudrun_specs: service => {
+      for secret, config in lookup(specs, "secrets", {}): secret => config
```

Comment:
What are your thoughts about if here, we make the default return a map with keys and null values. Then below, the conditions could be dropped?

Reply:
Are you suggesting something like this?:
```hcl
  cloudrun_secrets_attach = {
    for service, specs in local.cloudrun_specs: service => {
      for secret, config in lookup(specs, "secrets", {}): secrets => {
        version = lookup(config,"version",null)
        env_name = lookup(config,"env_name",null)
        mount_location = lookup(config,"mount_location",null)
        file_name = lookup(config,"file_name",null)
      }
    }
  }
```
I haven't tested if the module would work with null values for a defined block which has required values, so will test now. But think there would be errors, as the `version` is set to default to latest, or it might be specified in both volume mounts and environments, so the other values would be required.
Also, I feel that having the separate local variables makes things clearer in the [cloudrun module](https://github.com/mesoform/Multi-Cloud-Platform-Services/blob/b0d3bc3e0031c3fefeb935eb8eec4c32ca9a22df/gcp_cloudrun.tf#L71-L104)

Reply:
I meant something else actually but realise what I meant won't help much